src/read.c

```diff
@@ -4401,25 +4414,14 @@ static avifResult avifMetaFindAlphaItem(avifMeta * meta,
         }
     }
     assert(alphaItemCount == colorItemCount);
-    // Find an unused ID.
+    // Figure out the last used itemID.
     avifResult result;
-    if (meta->items.count >= UINT32_MAX - 1) {
-        // In the improbable case where all IDs are used.
+    const uint32_t lastID = meta->items.item[meta->items.count - 1]->id;
+    if (lastID == UINT32_MAX) {
+        // In the improbable case where the last ID is the maximum one, ids cannot be kept ordered.
```

Comment:
We should log a diagnostic message here. I thought of this in my previous review but somehow forgot to comment about it. Sorry.

Reply:
Done in https://github.com/AOMediaCodec/libavif/pull/1976